airflow/models/__init__.py

```diff
@@ -2812,16 +2821,39 @@ def get_default_view(self):
             return self.default_view
 
     @provide_session
-    def get_last_dagrun(self, session=None, include_externally_triggered=False):
-        return get_last_dagrun(self.dag_id, session=session,
-                               include_externally_triggered=include_externally_triggered)
+    def get_last_dagrun(self, include_externally_triggered=False, session=None):
+        """
+        Returns the last dag run for a dag, None if there was none.
+        Last dag run can be any type of run eg. scheduled or backfilled.
+        Overridden DagRuns are ignored.
+        """
+        return DagRun.get_last_dagrun(self.dag_id,
+                                      include_externally_triggered=include_externally_triggered,
+                                      session=session)
 
     @property
     def safe_dag_id(self):
         return self.dag_id.replace('.', '__dot__')
 
     def get_dag(self):
-        return DagBag(dag_folder=self.fileloc).get_dag(self.dag_id)
+        if self.is_subdag:
+            root_dag, path = self._get_nested_path_dag()
+            dagbag = DagBag(dag_folder=root_dag.fileloc)
+            dag = dagbag.get_dag(self.dag_id)
+        else:
+            dag = DagBag(dag_folder=self.fileloc).get_dag(self.dag_id)
+            if dag is None:
```

Comment:
```suggestion
        if dag is None:
```

This should apply to both subdags and plain dags

Reply:
+1

Reply:
agree there, changed this